typing/typecore.ml

```diff
@@ -3571,7 +3604,7 @@ and type_expect_
         type_application env loc expected_mode position funct funct_mode sargs
       in
       end_def ();
-      unify_var env (newvar()) funct.exp_type;
+      unify_var env (newvar Type_layout.any) funct.exp_type;
```

Comment:
I agree.  I'm basically leaving these untouched for now, but I think it would be an improvement if all the places where we unify with a fresh variable for levels reasons are factored out into one function with a comment explaining that.  (Which should of course be its own PR)